### ISSUE TITLE
Async calls in onSubmit do not update the $submitting store before completion

### ISSUE THREAD

**Issue by Kalos-S** [ciscoheat/sveltekit-superforms#284]
When using an async onSubmit hook the $submitting store of the form is not updated when the async function is called but only after it completes. This causes the form to potentially be submitted multiple times.

**Comment by ciscoheat** [comment]
This is a bit tricky - since you can cancel the submission in `onSubmit`, and the actual request isn't made until `onSubmit` completes, is the form submitting before that? For a programmer, I'd say no. But for the user, yes, since they clicked the button. So to uphold the illusion for the user, maybe you're right.

**Comment by shellicar** [comment]
@ciscoheat 
This seems to have broken SPA behaviour.

```ts
const options: FormOptions<FormSchema, any> = {
  SPA: true,
  dataType: "json",
  applyAction: false,
  taintedMessage: null,
  validators: formSchema,
  onUpdate: async ({ form, cancel }) => {
    const x = formSchema.safeParse(form.data);
    if(!x.success) {
      console.log(x.error.issues);
    } else {
      if (!await submitForm(x.data)) {
        cancel();
      }
    }
  },
  onUpdated: async({ form }) => {
    if (form.posted) {
      successAction();
    }
  },
};
```

I have beene using this pattern and noticed when updating to 1.10.2 submitting was no longer enabling (in reality it went to true then false immediately).
Downgrading to 1.10.1 fixed it for me.

Happy to change the way I am validating and submitting data, but trying to use onSubmit didn't work for me.

Let me know if I should open a new issue.


PS. Thanks for this great library.

**Comment by ciscoheat** [comment]
You're right, the timers are cancelling too early. It will be fixed in the next release.

**Comment by ciscoheat** [comment]
Fixed now with 1.11.0.
